code/gui/griddisplay.cpp

```diff
@@ -154,25 +154,134 @@ void GridDisplay::hide_grid() {
 
 void GridDisplay::selectRobotPosition(QString weight_selected) {
     // TODO: Use enum instead of strings
-    // TODO: simplify boolean expressions since there's a lot of branching
-    if (weight_selected == "Start Configuration") {
-        m_start_pos_selected = true;
-        m_end_pos_selected = false;
-    } else if (weight_selected == "End Configuration") {
-        m_start_pos_selected = false;
-        m_end_pos_selected = true;
-    } else {
-        m_start_pos_selected = false;
-        m_end_pos_selected = false;
-    }
+    m_start_pos_selected = (weight_selected == "Start Configuration");
+    m_end_pos_selected = (weight_selected == "End Configuration");
+//    if (weight_selected == "Start Configuration") {
+//        m_start_pos_selected = true;
+//        m_end_pos_selected = false;
+//    } else if (weight_selected == "End Configuration") {
+//        m_start_pos_selected = false;
+//        m_end_pos_selected = true;
+//    } else {
+//        m_start_pos_selected = false;
+//        m_end_pos_selected = false;
+//    }
 }
 
 void GridDisplay::init_start_end_pos() {
-    m_start_position.x = NOT_SELECTED_WEIGHT;
-    m_start_position.y = NOT_SELECTED_WEIGHT;
-    m_end_position.x = NOT_SELECTED_WEIGHT;
-    m_end_position.y = NOT_SELECTED_WEIGHT;
+    set_coordinates(m_start_position, NOT_SELECTED_WEIGHT, NOT_SELECTED_WEIGHT);
+    set_coordinates(m_end_position, NOT_SELECTED_WEIGHT, NOT_SELECTED_WEIGHT);
+}
+
+void GridDisplay::mouse_clicked() {
+
+}
+
+void GridDisplay::mousePressEvent(QMouseEvent *ev) {
+    qDebug() << "Mouse pressed: " << m_mouse_click_start.x << ", " << m_mouse_click_start.y << endl;
+    QWidget::mousePressEvent(ev);
+    select_start = QPoint(m_mouse_click_start.x, m_mouse_click_start.y);
+
+    rubber_band = new QRubberBand(QRubberBand::Rectangle, this);
+    rubber_band->setGeometry(QRect(ev->pos(), QSize()));
+    rubber_band->show();
+}
+
+void GridDisplay::mouseReleaseEvent(QMouseEvent *ev) {
+    qDebug() << "Mouse released: " << m_mouse_click_release.x << ", " << m_mouse_click_release.y << endl;
+    QWidget::mouseReleaseEvent(ev);
+    rubber_band->hide();
+    rect_select_buttons(m_mouse_click_start, m_mouse_click_release);
+}
+
+void GridDisplay::mouseMoveEvent(QMouseEvent *ev) {
+    qDebug() << "Mouse move event: " << ev->pos() << endl;
+    QWidget::mouseMoveEvent(ev);
+    rubber_band->setGeometry(QRect(select_start, ev->pos()).normalized());
+}
+
+void GridDisplay::paintEvent(QPaintEvent *ev) {
+//    QPainter painter(this);
+//    QRect rect = ev->rect();
+//    painter.drawLine()
+}
+
+void GridDisplay::swap_rect_coords(int &x1, int &y1, int &x2, int &y2) {
```

Comment:
this doesn't need to be a member function of GridDisplay

declare it at the top as `static void swap_rect_coords(...)`